src/_guides/language/language-tour.md

```diff
@@ -3547,6 +3554,65 @@ enum Color { red, green, blue }
   to help prevent copy-paste errors.
 {{site.alert.end}}
 
+#### Declaring enhanced enums
+
+Dart also allows enum declarations to declare classes
+with fields, methods, and const constructors
+which are limited to a fixed number of known constant instances.
+
+To declare an enhanced enum,
+follow a syntax similar to normal [classes](#classes),
+but with a few extra requirements:
+
+* Instance variables must be `final`, 
+  including those added by [mixins](#mixins).
+* All constructors must be
+  [constant generative constructors](#constant-constructors).
```

Comment:
Well, actually: 
  
 All generative constructors must be constant.
 All factory constructors must be non-constant (and can only return one of the fixed, known enum instances)..

You can have redirecting generative constructors (but they must be constant and redirect to another generative constructor until hitting a constant non-redirecting generative constructor).
And you can have factory constructors (which cannot be `const`, because even if redirecting, they are not allowed to redirect to a generative constructor).